consensus/state.go

```diff
@@ -596,6 +603,10 @@ func (cs *ConsensusState) receiveRoutine(maxSteps int) {
 		var mi msgInfo
 
 		select {
+		case rs_ := <-cs.txsAvailable:
+			// use nil for this special internal message signalling txs are available.
+			// no need to write this to the wal
+			cs.handleMsg(msgInfo{nil, ""}, rs_)
```

Comment:
msgInfo{nil, ""} -> actual dedicated message type.
remove rs_
remove rs in other handleMsg funcs.